codecarbon/external/hardware.py

```diff
@@ -67,6 +67,23 @@ def from_utils(cls, gpu_ids: Optional[List] = None) -> "GPU":
 
 @dataclass
 class CPU(BaseHardware):
-    @property
+    def __init__(self, output_dir: str):
+        self._output_dir = output_dir
+        self._intel_power_gadget = IntelPowerGadget(self._output_dir)
+
+    def _get_power_from_cpus(self) -> Power:
+        """
+        Get CPU power from Intel Power Gadget
+        :return: power in kW
+        """
+        all_cpu_details: Dict = self._intel_power_gadget.get_cpu_details()
+
+        power = 0
+        for metric, value in all_cpu_details.items():
+            if re.match("^Processor Power_\d+\(Watt\)$", metric):
+                power += value
+        logger.info(f"CODECARBON CPU Power Consumption : {power}")
```

Comment:
I'm guessing that this message will be logged everytime the energy is measure, right? If that's the case, we likely don't want the log message to be printed by default.

```suggestion
        logger.debug(f"CODECARBON CPU Power Consumption : {power}")
```

Reply:
yes, that is essentially the goal